### PR TITLE
introduce G4e refitter in the tracker alignment machinery

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

This PR has 3 commits:
   * 63c2db620d9ae9b1f05bead9a4e85ed7780b9985 introduces the machinery to be able to run the Geant4e refit in the `trackselectionRefitting` code, that is consumed by all the alignment and validation algorithms.
   * 0ea34e981c79f86e775bc0b10821048188a420da adds a dedicated unit test for this feature, now that https://github.com/cms-sw/cmssw/pull/38864 has fixed the core algorithm to be able to run in recent releases.
   * 00190045d3005f83ad0ebd86087b50c6afa9319c is a leftover from https://github.com/cms-sw/cmssw/pull/39013. In this commit I supply coverage for the track collections introduced in PR https://github.com/cms-sw/cmssw/pull/33770.

#### PR validation:

Relies on the augmented unit tests

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Not a backport, but shall be backported.

### CODE REVIEW

**Comment by mmusich** [comment]
type trk, new-feature

**Comment by mmusich** [comment]
please test

File: Alignment/CommonAlignment/python/tools/trackselectionRefitting.py [hunk 1]
```diff
@@ -267,6 +289,17 @@ def getSequence(process, collection,
...
                 ("TrackRefitter", "Second", {"method": "load",
                                              "clone": True})]
+    elif g4Refitting:
+        mods = [("TrackSelector", "HighPurity", {"method": "import"}),
+                ("TrackRefitter", "First", {"method": "load",
+                                            "clone": True}),
+                ("TrackHitFilter", "Tracker", {"method": "load"}),
+                ("TrackFitter", "HitFilteredTracks", {"method": "import"}),
+                ("TrackSelector", "Alignment", {"method": "load"}),
+                #("geopro","", {"method": "load"}),
```
Comment (tvami) [inline Alignment/CommonAlignment/python/tools/trackselectionRefitting.py:299]: is this supposed to be commented out?

Comment (mmusich) [reply Alignment/CommonAlignment/python/tools/trackselectionRefitting.py:299]: for the moment, yes

Comment (tvami) [reply Alignment/CommonAlignment/python/tools/trackselectionRefitting.py:299]: ok

**Comment by tvami** [comment]
+alca
 * unit tests pass

**Comment by qliphy** [comment]
+1

**Pull request merged**